src/test/java/uk/gov/pay/api/service/GetPaymentRefundsServiceTest.java

```diff
@@ -0,0 +1,81 @@
+import uk.gov.pay.commons.testing.pact.consumers.Pacts;
+
+import javax.ws.rs.client.Client;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GetPaymentRefundsServiceTest {
+
+    private static final String ACCOUNT_ID = "123456";
+    @Rule
+    public PactProviderRule ledgerRule = new PactProviderRule("ledger", this);
+    @Mock
+    private PublicApiConfig mockConfiguration;
+    @Mock
+    private ConnectorService mockConnectorService;
+    private GetPaymentRefundsService getPaymentRefundsService;
+
+    @Before
+    public void setUp() {
+        when(mockConfiguration.getLedgerUrl()).thenReturn(ledgerRule.getUrl());
+        when(mockConfiguration.getBaseUrl()).thenReturn("http://publicapi.test.localhost/");
+
+        PublicApiUriGenerator publicApiUriGenerator = new PublicApiUriGenerator(mockConfiguration);
+        LedgerUriGenerator ledgerUriGenerator = new LedgerUriGenerator(mockConfiguration);
+
+        Client client = RestClientFactory.buildClient(new RestClientConfig(false));
+        LedgerService ledgerService = new LedgerService(client, ledgerUriGenerator);
+
+        getPaymentRefundsService = new GetPaymentRefundsService(mockConnectorService, ledgerService, publicApiUriGenerator);
+    }
+
+    @Test
+    @PactVerification("ledger")
+    @Pacts(pacts = {"publicapi-ledger-get-payment-refunds"})
+    public void shouldReturnRefundsForPaymentCorrectlyFromLedger() {
+        Account account = new Account(ACCOUNT_ID, TokenPaymentType.CARD);
+        RefundsResponse response = getPaymentRefundsService.getLedgerTransactionTransactions(account, "ch_123abc456xyz");
+        assertThat(response.getPaymentId(), is("ch_123abc456xyz"));
+        assertThat(response.getLinks().getSelf().getHref(), is("http://publicapi.test.localhost/v1/payments/ch_123abc456xyz/refunds"));
+        assertThat(response.getLinks().getPayment().getHref(), is("http://publicapi.test.localhost/v1/payments/ch_123abc456xyz"));
+
+        List<RefundResponse> refunds = response.getEmbedded().getRefunds();
+        assertThat(refunds.size(), is(2));
+
+        assertThat(refunds.get(0).getRefundId(), is("refund-transaction-id1"));
+        assertThat(refunds.get(0).getStatus(), is("success"));
+        assertThat(refunds.get(0).getAmount(), is(1000L));
+        assertThat(refunds.get(0).getCreatedDate(), is("2018-09-22T10:14:16.067Z"));
+        assertThat(refunds.get(0).getLinks().getPayment().getHref(), is("http://publicapi.test.localhost/v1/payments/ch_123abc456xyz"));
+        assertThat(refunds.get(0).getLinks().getSelf().getHref(), is("http://publicapi.test.localhost/v1/payments/ch_123abc456xyz/refunds/refund-transaction-id1"));
+
+        assertThat(refunds.get(1).getRefundId(), is("refund-transaction-id2"));
+        assertThat(refunds.get(1).getStatus(), is("success"));
```

Comment:
Would be good to have unsuccessful one as well.

Reply:
👍 Will update ledger state and then make changes here.

Reply:
Updated https://github.com/alphagov/pay-publicapi/blob/149e2c833dc3e0af77128f355fb4e27ddf20cd20/src/test/resources/pacts/publicapi-ledger-get-payment-refunds.json#L50